File: docs/algorithms/code/algorithms/isAnagram.ts
function isAnagram(s: string, t: string): boolean {
  const map: Record<string, number> = {};

  for (let i = 0; i < s.length; i++) {
    if (map[s[i]]) {
      map[s[i]] += 1;
    } else {
      map[s[i]] = 1;
    }
  }

  for (let i = 0; i < t.length; i++) {
    if (!map[t[i]]) {
      return false;
    }
    map[t[i]] -= 1;
  }

  for (let i = 0; i < s.length; i++) {
    if (map[s[i]]) {
      return false;
    }
  }

  return true;
}

File: docs/algorithms/code/algorithms/isAnagram.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	hashMap := make(map[rune]int)

	for _, char := range s {
		hashMap[char]++
	}

	for _, char := range t {
		curr, ok := hashMap[char]

		if !ok || curr == 0 {
			return false
		}

		hashMap[char]--
	}

	return true
}